paper/src/main/java/com/denizenscript/denizen/paper/properties/PaperEntityExtensions.java

```diff
@@ -0,0 +1,171 @@
+package com.denizenscript.denizen.paper.properties;
+
+import com.denizenscript.denizen.nms.NMSHandler;
+import com.denizenscript.denizen.nms.NMSVersion;
+import com.denizenscript.denizen.objects.EntityFormObject;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.MapTag;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.ExperienceOrb;
+import org.bukkit.entity.Goat;
+import org.bukkit.inventory.EquipmentSlot;
+
+import java.util.UUID;
+
+public class PaperEntityExtensions {
+
+    public static void register() {
+
+        // <--[tag]
+        // @attribute <EntityTag.spawn_reason>
+        // @returns ElementTag
+        // @group properties
+        // @Plugin Paper
+        // @description
+        // Returns the entity's spawn reason.
+        // Valid spawn reasons can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/entity/CreatureSpawnEvent.SpawnReason.html>
+        // -->
+        EntityTag.registerSpawnedOnlyTag(ElementTag.class, "spawn_reason", (attribute, entity) -> {

```

Comment:
Does this actually require the entity to be spawned? the old tag didn't have that check, and don't see a reason it wouldn't work on e.g. an entity that used to exist but despawned and is still stored somewhere (it might require that, just test to make sure, as the old impl didn't require that)